Makefile

```diff
@@ -55,24 +72,86 @@ starknet-pause:
 starknet-unpause:
 	@. ./contracts/starknet/.env && ./contracts/starknet/change_pause_state.sh unpause
 
-test: 
-	make starknet-test
-	make ethereum-test
 
-.ONESHELL:
-starknet-deploy-and-connect: starknet-build
-	@. ./contracts/ethereum/.env && . ./contracts/starknet/.env
-	@. ./contracts/starknet/deploy.sh
-	@. ./contracts/ethereum/set_escrow.sh
-	@. ./contracts/ethereum/set_claim_payment_selector.sh
+### ZKSYNC ###
+
+zksync-clean:
+	@cd ./contracts/zksync/ && yarn clean
+
+zksync-build: zksync-clean
+	@cd ./contracts/zksync/ && yarn compile
+
+zksync-deploy: 
+	@make zksync-build && \
+	. ./contracts/zksync/.env && . ./contracts/zksync/deploy.sh
+
+zksync-connect:
+	@. ./contracts/ethereum/.env && . ./contracts/zksync/.env && \
+	. ./contracts/ethereum/set_zksync_escrow.sh
+
+zksync-deploy-and-connect: zksync-build
+	@. ./contracts/ethereum/.env && . ./contracts/zksync/.env && \
+	. ./contracts/zksync/deploy.sh && \
+	. ./contracts/ethereum/set_zksync_escrow.sh
+
+
+zksync-test: zksync-build
+	@cd ./contracts/zksync/ && yarn test
 
+#wip:
+.ONESHELL:
+zksync-test-integration:
+	@make ethereum-build && make zksync-build && \
+	. ./contracts/ethereum/test/.env.test && . ./contracts/zksync/test/.env.test && \
+	. ./contracts/ethereum/deploy.sh && \
+	. ./contracts/zksync/deploy.sh && \
+	. ./contracts/ethereum/set_zksync_escrow.sh && \
+	. ./contracts/zksync/test/set_order.sh && \
+	. ./contracts/zksync/test/transfer.sh && \
+	. ./contracts/zksync/test/claim_payment.sh
+
+# . ./setOrder - transfer - claim payment
+
+# zksync-upgrade: WIP
+
+
+### MULTI ###
+
+ethereum-and-zksync-deploy:
```

Comment:
It was implemented this way because using .ONESHELL and executing one line at a time was having unpredictable behaviour with bugs I could not fix, regarding variable exporting.

This way is still clean, tidy, and has more atomic and robust behaviour.

Reply:
refactored make targets to new "&& \" format, instead of ONESHELL as new format has been more reliable